massa-execution-worker/src/interface_impl.rs

```diff
@@ -1175,6 +1198,131 @@ impl Interface for InterfaceImpl {
         let hash = massa_hash::Hash::compute_from(bytes);
         Ok(hash.into_bytes())
     }
+
+    fn init_call_wasmv1(&self, address: &str, raw_coins: NativeAmount) -> Result<Vec<u8>> {
+        unimplemented!("init_call")
+    }
+
+    /// Returns a NativeAmount from a string
+    fn native_amount_from_str_wasmv1(&self, amount: &str) -> Result<NativeAmount> {
+        let amount = Amount::from_str(amount).map_err(|err| anyhow!(format!("{}", err)))?;
+        let amount = amount_to_native_amount(&amount).map_err(|err| anyhow!(format!("{}", err)))?;
+
+        Ok(amount)
+    }
+
+    /// Returns a string from a NativeAmount
+    fn native_amount_to_string_wasmv1(&self, amount: &NativeAmount) -> Result<String> {
+        let amount = amount_from_native_amount(amount)
+            .map_err(|err| anyhow!(format!("Couldn't convert native amount to Amount: {}", err)))?;
+
+        Ok(amount.to_string())
+    }
+
+    /// Checks if the given native amount is valid
+    fn check_native_amount_wasmv1(&self, amount: &NativeAmount) -> Result<bool> {
+        Ok(amount_from_native_amount(amount).is_ok())
+    }
+
+    /// Adds two native amounts, saturating at the numeric bounds instead of overflowing.
+    fn add_native_amounts_wasmv1(
+        &self,
+        amount1: &NativeAmount,
+        amount2: &NativeAmount,
+    ) -> Result<NativeAmount> {
+        let amount1 = amount_from_native_amount(amount1)?;
+        let amount2 = amount_from_native_amount(amount2)?;
+        let sum = amount1.saturating_add(amount2);
+        let sum = amount_to_native_amount(&sum)
+            .map_err(|err| anyhow!(format!("Couldn't add native amounts: {}", err)))?;
+        Ok(sum)
+    }
+
+    /// Subtracts two native amounts, saturating at the numeric bounds instead of overflowing.
+    fn sub_native_amounts_wasmv1(
+        &self,
+        amount1: &NativeAmount,
+        amount2: &NativeAmount,
+    ) -> Result<NativeAmount> {
+        let amount1 = amount_from_native_amount(amount1)?;
+        let amount2 = amount_from_native_amount(amount2)?;
+        let sub = amount1.saturating_sub(amount2);
+        let sub = amount_to_native_amount(&sub)
+            .map_err(|err| anyhow!(format!("Couldn't sub native amounts: {}", err)))?;
+        Ok(sub)
+    }
+
+    /// Multiplies a native amount by a factor, saturating at the numeric bounds instead of overflowing.
+    fn mul_native_amount_wasmv1(&self, amount: &NativeAmount, factor: u64) -> Result<NativeAmount> {
+        let amount = amount_from_native_amount(amount)?;
+        let mul = amount.saturating_mul_u64(factor);
+        let mul = amount_to_native_amount(&mul)
+            .map_err(|err| anyhow!(format!("Couldn't mul native amount: {}", err)))?;
+        Ok(mul)
+    }
+
+    /// Divides a native amount by a divisor, return an error if the divisor is 0.
+    fn div_rem_native_amount_wasmv1(
```

Comment:
I think it would be better to add this to the Amount structure:

```rust
pub fn checked_rem(&self, other: &Amount) -> Option<Amount> {
    Amount(self.0.checked_rem(other.0)?)
}

pub fn checked_rem_u64(&self, other: u64) -> Option<Amount> {
    Amount(self.0.checked_rem(other)?)
}

```

Reply:
many thanks, will use it after I've synced with some dependencies